.github/workflows/ci.yml

```diff
@@ -0,0 +1,352 @@
+name: CI 
+
+on:
+  push:
+    paths:
+      - '.github/workflows/dart_ci.yml'
+      - 'discovery_engine/**'
+      - 'discovery_engine_core/**'
+      - 'job_runner.sh'
+      - '.github/workflows/rust_ci.yml'
+      - 'discovery_engine_core/**'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+env:
+  DART_VERSION: '2.14.4'
+  DART_WORKSPACE: ${{ github.workspace }}/discovery_engine
+  RUST_WORKSPACE: ${{ github.workspace }}/discovery_engine_core
+  RUST_NIGHTLY: nightly-2021-09-09
+  RUSTFLAGS: "-D warnings"
+  RUSTDOCFLAGS: "-D warnings"
+
+
+jobs:
+  selection:
+    runs-on: ubuntu-latest
+    # Set job outputs to values from filter step
+    outputs:
+      dart: ${{ steps.filter.outputs.dart }}
+      rust: ${{ steps.filter.outputs.rust }}
+    steps:
+      # For pull requests it's not necessary to checkout the code
+      - uses: dorny/paths-filter@v2.10.2
+        id: filter
+        with:
+          filters: |
+            dart:
+              - '.github/workflows/dart_ci.yml'
+              - 'discovery_engine/**'
+              - 'discovery_engine_core/**'
+              - 'job_runner.sh'
+            rust:
+              - '.github/workflows/rust_ci.yml'
+              - 'discovery_engine_core/**'
+
+  # DART CI 
+  dart-format:
+    name: dart-format
+    needs: selection
+    if: ${{ needs.changes.outputs.dart == 'true' }}
+    runs-on: ubuntu-20.04
+    timeout-minutes: 10
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+
+      - name: Install dart
+        uses: dart-lang/setup-dart@6a218f2413a3e78e9087f638a238f6b40893203d # v1.3
+        with:
+          sdk: ${{ env.DART_VERSION }}
+
+      - name: Check formatting
+        working-directory: ${{ env.DART_WORKSPACE }}
+        run: dart format --output=none --set-exit-if-changed .
+
+  dart-analyze:
```

Comment:
as in rust we can have all dart-* run in parallel. Don't know how easy it is but would be nice if `dart-test` could depend on `rust-test` if it must run.

Reply:
I'm going to create a new task for this.